src/connection_manager.c

```diff
@@ -763,12 +775,28 @@ QD_EXPORT qd_connector_t *qd_dispatch_configure_connector(qd_dispatch_t *qd, qd_
         DEQ_INSERT_TAIL(cm->connectors, ct);
         log_config(cm->log_source, &ct->config, "Connector");
 
+        qdr_core_t* core = qd_dispatch_router_core(qd);
+        int connection_count = 0;
+        //
+        // If the user asks for automatic setting of the number
+        // of data connnection count, make one data connection
+        // for every two worker threads. This is the best ratio
+        // for high load, as determined by throughput performance
+        // testing.
+        //
+        if (!strcmp("auto", ct->config.data_connection_count)) {
+          connection_count = (qdr_core_get_worker_thread_count(core) + 1) / 2;
+          qd_log(cm->log_source, QD_LOG_INFO, "Inter-router data connections calculated at %d ", connection_count);
+        } else {
+          connection_count = atoi(ct->config.data_connection_count);
```

Comment:
Beware - atoi() doesn't check input. Things like atoi("gerbil") returns zero. You might be better off using sscanf(data_connection_count, "%u", &unsigned_var) and verifying the return code is 1.

That's the gotcha with using a string type for configuration input: you need to validate the input manually.

If you do want to sanitize the input then dealing with the error would be simple: just call qd_error(QD_ERROR_CONFIG, "some error message") then goto error;  At least I think that should work....

Reply:
I made an interesting (to me, anyway) change here.  We'll see what you think.